src/main.ts

```diff
@@ -25,12 +25,16 @@ export async function run(
         dependencies.endGroup()
 
         dependencies.startGroup(MESSAGES.STEP_LABELS.RUNNING_CODE_ANALYZER)
-        const codeAnalyzerExitCode: number = await commandExecutor.runCodeAnalyzer(
+        const codeAnalyzerOutput: CommandOutput = await commandExecutor.runCodeAnalyzer(
             inputs.runCommand,
             inputs.runArgs,
             INTERNAL_OUTFILE
         )
-        dependencies.setOutput('exit-code', codeAnalyzerExitCode.toString())
+        dependencies.setOutput('exit-code', codeAnalyzerOutput.exitCode.toString())
+        if (codeAnalyzerOutput.exitCode !== 0 && codeAnalyzerOutput.stderr.includes('Error')) {
```

Comment:
nitpick: Can "Error" be a string constant since it's used multiple times? The case sensitivity also looks tricky.

Reply:
Yes, since this is a nice to have, i'm leaning on how sf returns errors... it currently always has "Error (\d+)" and I don't want to pick up warnings if possible. So this I think is the safest thing to do and the worse case scenario is that it doesn't satisfy the conditional code and the error is still just in the logs. So I think this is fine. This is something we should consider moving forward... having warnings and errors mixed in stderr kind of prevents us from capturing errors... which is another reason why calling a cli instead of having a direct dependency is tricky.